lib/fluent/config/configure_proxy.rb

```diff
@@ -103,6 +112,9 @@ def merge(other) # self is base class, other is subclass
 
         merged = self.class.new(other.name, options)
 
+        # configured_in MUST be kept
+        merged.configured_in_section = self.configured_in_section
+
```

Comment:
Because it is not for plugin itself, but for internal handling about overwriting default values. It should be separated from argument/params/defaults/sections.